### PR TITLE
feat(salmon): download and build for salmon v1.4.0

### PR DESCRIPTION
**Pull request by jemten**
### This PR adds | fixes:

- download gencode transcript sequences
- build salmon index 

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: lib/MIP/Recipes/Build/Salmon_quant_prerequisites.pm [hunk 1]
```diff
@@ -206,24 +208,60 @@ sub build_salmon_quant_prerequisites {
...
+        {
+            filehandle   => $filehandle,
+            infile_path  => $decoy_file_path,
+            inplace_edit => 1,
+            script       => q{'s/>//g'},
+        }
+    );
+    say {$filehandle} $NEWLINE;
+
+    ## Combine transcriptome and genome
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Build/Salmon_quant_prerequisites.pm:242]: Wow how interesting!

File: lib/MIP/Recipes/Build/Salmon_quant_prerequisites.pm [hunk 2]
```diff
@@ -206,24 +208,60 @@ sub build_salmon_quant_prerequisites {
...
-            fasta_path   => catfile( $salmon_quant_directory_tmp, q{cDNA_seqs.fa} ),
-            filehandle   => $filehandle,
-            outfile_path => $salmon_quant_directory_tmp,
+            fasta_path     => catfile( $salmon_quant_directory_tmp, q{gentrome.fa} ),
+            threads        => $NUMBER_OF_CORES,
+            filehandle     => $filehandle,
+            temp_directory => $temp_directory,
+            gencode        => $active_parameter_href->{salmon_quant_gencode_reference},
+            decoy_path     => $decoy_file_path,
+            outfile_path   => $salmon_quant_directory_tmp,
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Build/Salmon_quant_prerequisites.pm:264]: Order AB

**Review by henrikstranneheim — APPROVED** [review]
Looks good!
There was some github annotations taht needs fixing

File: lib/MIP/Recipes/Build/Salmon_quant_prerequisites.pm (hunk 1 above)
Comment (jemten) [reply lib/MIP/Recipes/Build/Salmon_quant_prerequisites.pm:242]: Yeah, can't say I fully understand the salmon mapping 😆

**Pull request merged**